src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationMiddleware.cs

```diff
@@ -125,6 +128,20 @@ public class OpenIdConnectAuthenticationMiddleware : AuthenticationMiddleware<Op
                     Options.ConfigurationManager = new ConfigurationManager<OpenIdConnectConfiguration>(Options.MetadataAddress, httpClient);
                 }
             }
+
```

Comment:
Silent failure/fallback for misconfiguration is hard to debug. I'd expect something more like:
```
if  (Options.CacheNonces && Options.NonceCache == null)
{
  Options.NonceCache = services.GetRequiredService<IDistributedCache>(); // throws
}
```
Or maybe GetServices and check for null to throw your own error.